contracts/FlowCallbackScheduler.cdc

```diff
@@ -802,97 +793,168 @@ access(all) contract FlowCallbackScheduler {
+                    }
+                }
+
+                pendingCallbacks = pendingCallbacks
+                    .concat(high)
+                    .concat(medium)
+                    .concat(low)
+            }
+
+            return pendingCallbacks
+        }
+
+        /// removeExecutedCallbacks removes all callbacks that are marked as executed.
+        access(Process) fun removeExecutedCallbacks() {
+            let currentTimestamp = getCurrentBlock().timestamp
+            let pastTimestamps = self.sortedTimestamps.getBefore(current: currentTimestamp)
+
+            for timestamp in pastTimestamps {
+                let callbackIDs = self.slotQueue[timestamp] ?? {}
+                
+                for id in callbackIDs.keys {
+                    let callback = self.borrowCallback(id: id)
+                        ?? panic("Invalid ID: \(id) callback not found during initial processing") // critical bug
+
+                    // Only remove executed callbacks
+                    if callback.status != Status.Executed {
+                        continue
                     }
+
+                    destroy self.removeCallback(callback: callback)
                 }
             }
         }
 
+        /// process scheduled callbacks and prepare them for execution. 
+        ///
+        /// It iterates over past timestamps in the queue and processes the callbacks that are 
+        /// eligible for execution. It also emits an event for each callback that is processed.
+        ///
+        /// This function is only called by the FVM to process callbacks.
+        access(Process) fun process() {
+            let pendingCallbacks = self.pendingQueue()
+            
+            if pendingCallbacks.length == 0 {
+                return
+            }
+
+            for callback in pendingCallbacks {
+                emit PendingExecution(
+                    id: callback.id,
+                    priority: callback.priority.rawValue,
+                    executionEffort: callback.executionEffort,
+                    fees: callback.fees.balance,
+                    callbackOwner: callback.handler.address
+                )
+
+                // after pending execution event is emitted we set the callback as executed because we 
+                // must rely on execution node to actually execute it. Execution of the callback which is 
+                // done in a separate transaction that calls executeCallback(id) function can not update 
+                // the status of callback or any other shared state, since that blocks concurrent callback 
+                // execution. Therefore optimistic update to executed is made here to avoid race condition.
+                callback.setStatus(newStatus: Status.Executed)
+
+                // charge the fee for callback execution
+                destroy callback.payAndRefundFees(refundMultiplier: 0.0)
+            }
+
+            self.removeExecutedCallbacks()
```

Comment:
This can't be done here. Remember, we can't remove callbacks until after they are actually executed because that also removes the callback resource, meaning there would be no callback to execute, so they can't be removed in the same transaction that they are processed. The easiest change here would be to just put the call to `removeExecutedCallbacks before the call to `pendingQueue`